examples/openthread/hello/README.md

```diff
@@ -0,0 +1,231 @@
+# OpenThread Hello 
+
+The openthread_hello application serves to demonstrate a working, albeit, incomplete 
+implementation of OpenThread in libtock-c. As currently configured, this app successfully
+configures, joins, and remains attached to a Thread network. Furthermore this app is 
+able to send and receive UDP packets over the network. This README serves as a guide for
+setting up a Thread network for this app to join, describes the current build system 
+used, and details the numerous TODOs (and shortcomings) of the current implementation/build
+system. The __OpenThread Port for Tock__ subsection provides context while the __Thread Network
+Setup__ and __Tock OpenThread Setup__ provide "how to" guides for setting up the network
+for the Tock device to attach to and the Tock device as well. This app will only work when using 
+the nrf52840dk board.
+
+## OpenThread Port for Tock
+
+### Overview
+The Tock/OpenThread port is implemented in the `libtock-c/libopenthread` directory. OpenThread 
+utilizes a CMake build system. To port OpenThread to Tock, we alter the compile arguments 
+provided to OpenThread, initiate the OpenThread build system, and then use the compiled 
+static library as an external library within the `libtock-c` `make` build system. 
+
+The OpenThread static library implements the logic needed to create, attach to, maintain,
+and operate within a Thread network. OpenThread utilizes a platform abstraction layer 
+that provides OpenThread with the needed "access" to hardware primitives (e.g. 
+IEEE 802.15.4 radio, TRNG, alarm, flash). The platform abstraction layer "connects"
+OpenThread to use the relevant `libtock-c` functionality. This is implemented in the
+`libtock-c/libopenthread/platform` directory.
+
+### TODOs and Shortcomings
+
+- The Tock Thread device disconnects from the network (i.e. does not send child update request 
+and subsequently times out) after ~7 minutes. This bug has been traced to an issue with the alarm
+implementation but has not been resolved.
+- The 6LoWPAN capsule on occassion panics (this behavior is observed semi frequently)
+- The code size is quite large as is for the current implementation. There are numerous 
+opportunities to reduce this (use hardware crypto instead of mbed crypto library).
+- The openthread_hello `Makefile` will build all needed OpenThread and platform dependencies. 
+However, the `make` build system, as is, does not rebuild the platform or OpenThread library 
+if changes are made to either the platform or openthread directories.
+- Channel switching is not yet implemented
+- The RSSI value is hard coded 
```

Comment:
I think these are all in the process of being resolved?

Reply:
Yes, with the exception of channel switching.

LQI/RSSI should be merged soon (https://github.com/tock/tock/pull/3972).
6LoWPAN bug is resolved.
The disconnect bug is mostly fixed (blocking on https://github.com/tock/tock/pull/3973)

Channel switching for now is not implemented (and in my opinion should not be a blocker on this PR. This is something we can add support for in the 15.4 capsule rewrite)